test/unit-tests/expression/function/evaluate.test.js

```diff
@@ -84,4 +84,101 @@ describe('evaluate', function () {
+      assert.strictEqual(math.evaluate('null ?? 2^3'), 8) // null ?? (2^3)
+
+      // ?? has same precedence as logical OR, left associative
+      assert.strictEqual(math.evaluate('null ?? false or true'), true) // (null ?? false) or true
+      assert.strictEqual(math.evaluate('true or null ?? 42'), true) // (true or null) ?? 42 = true ?? 42 = true
+
+      // Parentheses can override precedence
+      assert.strictEqual(math.evaluate('null ?? (1 + 2)'), 3)
+      assert.strictEqual(math.evaluate('(null ?? 1) + 2'), 3) // (null ?? 1) + 2 = 1 + 2 = 3
+    })
+
+    it('should handle nullish coalescing with complex expressions', function () {
+      const scope = { a: null, b: 5, c: 0 }
+      assert.strictEqual(math.evaluate('a ?? b * 2', scope), 10) // null ?? (5 * 2)
+      assert.strictEqual(math.evaluate('c ?? b * 2', scope), 0) // 0 ?? (5 * 2) = 0
+      assert.strictEqual(math.evaluate('(a ?? b) * 2', scope), 10) // (null ?? 5) * 2 = 10
+    })
+
+    it('should handle nullish coalescing with strings', function () {
+      assert.strictEqual(math.evaluate('null ?? "hello"'), 'hello')
+      assert.strictEqual(math.evaluate('"world" ?? "hello"'), 'world')
+      assert.strictEqual(math.evaluate('"" ?? "hello"'), '') // empty string is not nullish
+    })
+
+    it('should handle nullish coalescing with matrices and arrays', function () {
+      assert.deepStrictEqual(math.evaluate(['null ?? 1', '2 ?? null', 'null ?? null ?? 3']), [1, 2, 3])
+      assert.deepStrictEqual(math.evaluate(math.matrix(['null ?? 1', '2 ?? null'])), math.matrix([1, 2]))
+    })
+
+    it('should handle nullish coalescing with function calls', function () {
+      const scope = {
+        getValue: function () { return null },
+        getDefault: function () { return 42 }
+      }
+      assert.strictEqual(math.evaluate('getValue() ?? getDefault()', scope), 42)
+
+      const scope2 = {
+        getValue: function () { return 10 },
+        getDefault: function () { return 42 }
+      }
+      assert.strictEqual(math.evaluate('getValue() ?? getDefault()', scope2), 10)
+    })
+
+    it('should handle nullish coalescing with conditional expressions', function () {
+      // Conditional has lower precedence than nullish coalescing
+      assert.strictEqual(math.evaluate('true ? null ?? 5 : 10'), 5) // true ? (null ?? 5) : 10
+      assert.strictEqual(math.evaluate('false ? null ?? 5 : 10'), 10)
+      assert.strictEqual(math.evaluate('null ?? true ? 5 : 10'), 5) // (null ?? true) ? 5 : 10
+    })
+
+    it('should handle nullish coalescing with object property access simulation', function () {
+      const obj = { foo: 7, bar: null }
+      assert.strictEqual(math.evaluate('foo ?? 0', obj), 7)
+      assert.strictEqual(math.evaluate('bar ?? 0', obj), 0)
+      // baz is undefined, would throw error without fallback
+      assert.throws(() => math.evaluate('baz ?? 0', obj), /Undefined symbol baz/)
+    })
```

Comment:
This test appears to duplicate the one at lines 138-141 above. Did you mean to add a test where an operand to the nullish coalescing operator was an access to a property of an object? that would be something different from what you have above.